unlock-app/src/components/interface/locks/Settings/elements/EmailTemplatePreview.tsx

```diff
@@ -0,0 +1,209 @@
+})
+
+type FormSchemaProps = z.infer<typeof FormSchema>
+
+export const EmailTemplatePreview = ({
+  templateId,
+  disabled,
+  network,
+  lockAddress,
+}: EmailTemplatePreviewProps) => {
+  const config = useConfig()
+  const [showPreview, setShowPreview] = useState(false)
+  const wedlocksService = useWedlockService()
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isDirty },
+    watch,
+    setValue,
+    reset,
+  } = useForm<FormSchemaProps>()
+
+  const customContent = watch('customContent', '')
+  const onSaveCustomContent = async () => {
+    const saveEmailPromise = storage.saveCustomEmailContent(
+      network,
+      lockAddress,
+      templateId,
+      {
+        data: {
+          content: customContent,
+        },
+      }
+    )
+    await ToastHelper.promise(saveEmailPromise, {
+      loading: 'Updating custom email content...',
+      error: 'Could not update custom email content.',
+      success: 'Custom email content updated.',
+    })
+    reset({
+      customContent: customContent || '',
+      email: '',
+    })
+  }
+
+  const saveCustomContent = useMutation(onSaveCustomContent)
+
+  const onSubmit = async (form: FormSchemaProps) => {
+    const promise = wedlocksService.sendEmail(templateId as any, form.email)
+    await ToastHelper.promise(promise, {
+      loading: 'Sending email preview...',
+      success: 'Email preview sent.',
+      error: `Can't send email preview`,
+    })
+    setShowPreview(false) // close modal after email is sent
+  }
+
+  const [_data, { data: emailHtmlPreview }] = useQueries({
+    queries: [
+      {
+        queryKey: ['getCustomContent', network, lockAddress, templateId],
+        queryFn: async () => {
+          const res = await storage.getCustomEmailContent(
+            network,
+            lockAddress,
+            templateId
+          )
+          return res?.data?.content || ''
+        },
+        onSuccess: (content: any) => {
+          setValue('customContent', content || '')
+        },
+        enabled: !disabled,
+      },
+      {
+        queryKey: [
+          'getEmailPreview',
+          network,
+          lockAddress,
+          templateId,
+          saveCustomContent.isSuccess,
+        ],
+        queryFn: async () => {
+          const url = `${config.services.wedlocks.host}/preview/${templateId}`
+          const res = await (await fetch(url)).text()
+          return res
+        },
+      },
+    ],
+  })
+
+  const loading = saveCustomContent.isLoading
+  const disableShowPreview = loading || saveCustomContent.isLoading || isDirty
+
+  return (
+    <>
+      <div className="flex flex-col justify-start gap-3">
+        <div className="pb-2">
+          <TextBox
+            placeholder="## Example content"
+            rows={8}
+            {...register('customContent')}
+          />
+          <div className="pb-2 text-sm text-gray-700">
+            Markdown is supported for custom email content
```

Comment:
Please add link to the Markdown cheat sheet

Reply:
<img width="911" alt="Screenshot 2023-02-10 at 17 50 36" src="https://user-images.githubusercontent.com/20865711/218149560-46d2c5cd-d875-45ef-ae0f-a815b0675b53.png">